CANTroller2/src/devices.h

```diff
@@ -408,6 +472,30 @@ class AnalogSensor : public Sensor<NATIVE_T, HUMAN_T> {
         
 };
 
+// BatterySensor reads the voltage level from the Mule battery
+class BatterySensor : public AnalogSensor<int32_t, float> {
+    protected:
+        // NOTE: for now lets keep all the config stuff here in the class. could also read in values from a config file at some point.
```

Comment:
Agreed. Could also use a config object that reads from a config yaml or something in case we wanted to have different configs to choose from.
```
// Config struct 
struct BatteryConfig {
  float minVolts;
  float maxVolts;
  float initialAdc;
  // Other config values  
};

class BatterySensor {

  BatteryConfig config;
  
  BatterySensor(const BatteryConfig& batteryConfig) 
    : config(batteryConfig) {

    // Initialize limits, ADC values, etc from config

  }

}
```